### PR TITLE
dprint 0.11.0 (new formula)

### LINKED ISSUE #323: Homebrew support
👋  heads up, I have filed [a PR](https://github.com/Homebrew/homebrew-core/pull/65924) to introduce the homebrew support for this wonderful tool.

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

fixes https://github.com/dprint/dprint/issues/323
remake of #65924

### CODE REVIEW

File: Formula/dprint.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  homepage "https://dprint.dev/"
+  url "https://github.com/dprint/dprint/archive/0.11.0.tar.gz"
+  sha256 "beb0e57cf3db632a2f2704b7e0cfca49be5c2d4428d19fa10b62b9dfd6d4d210"
+  license "MIT"
+  head "https://github.com/dprint/dprint.git"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked", "--root", prefix, "--path", "./crates/dprint"
```
Comment (SMillerDev) [inline Formula/dprint.rb:12]: Isn't this `*std_cargo_args`?

Comment (carlocab) [reply Formula/dprint.rb:12]: Close, but not exactly:
```
def std_cargo_args
  ["--locked", "--root", prefix, "--path", "."]
end
```

Comment (SMillerDev) [reply Formula/dprint.rb:12]: Are we able to override the --path like we would for Go? I don't know anything about Cargo

Comment (carlocab) [reply Formula/dprint.rb:12]: Neither do I. If I were forced to do it, I'd do something like
```
args = std_cargo_args.reject { |s| s["."] } + ["./crates/dprint"]
system "cargo", "install", *args
```
but this is prone to breaking if someone decides to modify `std_cargo_args`.

Comment (SMillerDev) [reply Formula/dprint.rb:12]: If we can't override it I'm fine with the current version. It's worth a try though

Comment (carlocab) [reply Formula/dprint.rb:12]: Actually, better:
```suggestion
    args = std_cargo_args.map { |s| s == "." ? "./crates/dprint" : s }
    system "cargo", "install", *args
```

Comment (chenrui333) [reply Formula/dprint.rb:12]: I think I tried before, it did not work. Let me give another try.

**Pull request closed without merging**